src/rules/optimize-image/optimize-image.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * @fileoverview Optimize images.
+ */
+
+import { Category } from 'sonarwhal/dist/src/lib/enums/category';
+import { RuleContext } from 'sonarwhal/dist/src/lib/rule-context';
+import { IRule, IRuleBuilder, IScanEnd, ITargetFetchStart } from 'sonarwhal/dist/src/lib/types';
+import { debug as d } from 'sonarwhal/dist/src/lib/utils/debug';
+
+import { Result, File, Bytes } from './types';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+const rule: IRuleBuilder = {
+    create(context: RuleContext): IRule {
+        /** Error processing the request. */
+        let failed: boolean = false;
+        /** The promise that represents the analyzing result. */
+        let analyzePromise: Promise<Result>;
+
+        const username: string = process.env.SERVICE_USERNAME; // eslint-disable-line no-process-env
```

Comment:
Is there a reason why this isn't done through the rule configs?

Reply:
For now we want to do it via environment variables. We will be the only users of this rule for now while we test the endpoints.

Reply:
ok.